commcare_connect/opportunity/api/views.py

```diff
@@ -30,3 +44,24 @@ def get(self, *args, **kwargs):
             "total_modules": total_modules.count(),
         }
         return Response(UserLearnProgressSerializer(ret).data)
+
+
+class UserVisitVerificationStatus(APIView):
+    serializer_class = UserVisitVerificationStatusSerializer
+    permission_classes = [IsAuthenticated]
+
+    def get(self, *args, **kwargs):
+        user_visits = UserVisit.objects.filter(user=self.request.user, opportunity=kwargs.get("pk"))
+        total_visits = user_visits.count()
+        daily_visits = user_visits.filter(visit_date__date=datetime.today().date()).count()
+        approved = user_visits.filter(status=VisitValidationStatus.approved).count()
+        pending = user_visits.filter(status=VisitValidationStatus.pending).count()
+        rejected = user_visits.filter(status=VisitValidationStatus.rejected).count()
```

Comment:
you should be able to do these aggregations in a single query with conditional aggregations: https://docs.djangoproject.com/en/1.8/ref/models/conditional-expressions/#conditional-aggregation

Reply:
Would be good to put this is a tested function